### ISSUE TITLE
EventEmitter Memory leak

### ISSUE THREAD

**Issue by hardeep** [rauchg/slackin#131]
I'm getting the following error:

```
2016-01-11T20:31:03.523100206Z app[web.1]: (node) warning: possible EventEmitter memory leak detected. 11 change listeners added. Use emitter.setMaxListeners() to increase limit.
2016-01-11T20:31:03.526298663Z app[web.1]: Trace
2016-01-11T20:31:03.526330970Z app[web.1]:     at SlackData.addListener (events.js:239:17)
2016-01-11T20:31:03.526347370Z app[web.1]:     at Namespace.<anonymous> (/app/node_modules/slackin/node/index.js:177:11)
2016-01-11T20:31:03.526361474Z app[web.1]:     at emitOne (events.js:77:13)
2016-01-11T20:31:03.526374174Z app[web.1]:     at Namespace.emit (events.js:169:7)
2016-01-11T20:31:03.526386774Z app[web.1]:     at Namespace.emit (/app/node_modules/socket.io/lib/namespace.js:205:10)
2016-01-11T20:31:03.526409414Z app[web.1]:     at /app/node_modules/socket.io/lib/namespace.js:172:14
2016-01-11T20:31:03.526422117Z app[web.1]:     at doNTCallback0 (node.js:430:9)
2016-01-11T20:31:03.526435181Z app[web.1]:     at process._tickCallback (node.js:359:13)
```

The issue comes from the line: https://github.com/rauchg/slackin/blob/master/lib/index.js#L173

Every time a connection is made over a websocket a new listener is added to the object:

    let slack = new Slack({ token, interval, org });

You can confirm this by connecting to the slackin instance with 11 tabs.

https://nodejs.org/docs/latest/api/events.html#events_emitter_setmaxlisteners_n

Since this looks like an intended design I would like to propose adding either

    slack.setMaxListeners(0);

Or, a configurable variable to set the max listeners.

Should I go ahead and make a pull request for this?

Hardeep

**Comment by bennmapes** [comment]
+1

**Comment by hardeep** [comment]
I've submitted a pull request https://github.com/rauchg/slackin/pull/135

**Comment by hardeep** [comment]
resolved by pull request #135
